pkg/api/v1beta3/dynakube/oneagent_props.go

```diff
@@ -77,30 +84,20 @@ func (dk *DynaKube) OneAgentConnectionInfoConfigMapName() string {
 	return dk.Name + OneAgentConnectionInfoConfigMapSuffix
 }
 
-func (dk *DynaKube) NeedsReadOnlyOneAgents() bool {
-	return (dk.HostMonitoringMode() || dk.CloudNativeFullstackMode()) &&
-		dk.FeatureReadOnlyOneAgent()
+func (dk *DynaKube) NeedsCSIDriver() bool {
```

Comment:
Naming with `Needs`, `CanUse` and `Use` for the CSI driver is confusing to me and could lead to incorrect usage. Can we simplify this?

Reply:
It's a mess because it is kinda a mess in general :D 
- There is 1 feature (cloudNative) that will not work without the CSI-driver, so that should be validated
- There is 1 feature that can't use the csi-driver (classic), so you have to keep in mind
- There are 2 features (app/host-mon) that will use it if available, another thing to keep in mind

I tried (still trying) to just have an `IsCSIEnabled` check everywhere, but the problem is that you have to check more, as the features of cloudnative/classic/hostmon/appmon have shared code.
- Examples where using only `IsCSIEnabled` works
    - in the mutation webhook (csi is available => use it, easy)
- Examples where using only `IsCSIEnabled` doesn't work
    - oneagent-daemonset generation, because of classic, that is why we need `NeedsReadOnlyOneAgents`
    - in the validation webhook
       - the `disabledCSIForReadonlyCSIVolume` check requires the very elaborate `UseCSIDriver` check, because of the validation logic is a bit weird (don't really see why we even have this 🤷 )
       - the `isCSIModuleDisabled` check needs to know if the CSI-driver is a hard requirement or not 

I am tyring to come up with better names AND moving validation-specific logic to the validation package, and use an `IsCSIEnabled` check where possible